src/Plugin/Double.php

```diff
@@ -404,6 +404,10 @@ protected static function _generateMethod($method, $callParent = false)
     protected static function _generateReturnType($method)
     {
         $typehint = Inspector::returnTypehint($method->getReturnType());
+        $trimmed = ltrim($typehint, '\\');
+        if (in_array($trimmed, [ 'self', 'static' ])) {
```

Comment:
need to have strict check : 

```suggestion
        if (in_array($trimmed, [ 'self', 'static' ], true)) {
```